public/javascripts/webrtc.js

```diff
@@ -238,9 +256,59 @@ function handleIceCandidate(type, id, candidate) {
   });
 }
 
+function handleRtcPeerTrack(id) {
+  return function({ track, streams: [stream] }) {
+  console.log('Attempt to display peer media...');
+  displayStream(`#peer ID: ${id}`, stream);
+  }
+}
+
+//possible need for handleRtcConnectionStateChange
+
 /**
 David start
 */
+
+async function requestUserMedia(media_constraints) {
+  $self.stream = new MediaStream();
+  $self.media = await navigator.mediaDevices
+    .getUserMedia(media_constraints);
+  $self.stream.addTrack($self.media.getTracks()[0]);
+  displayStream('#self', $self.stream);
+}
+
+function createVideoElement(id) {
+  const figure = document.createElement('figure');
+  const figcaption = document.creaetElement('figcaption');
+  const video = document.createElement('video');
+  const video_attributes = {
+    'autoplay': '',
+    'playsinline': '',
+    'poster': '../images/placeholder.jpg'
+  };
+
+  figure.id = id;
+  figcaption.innerText = id;
+  for (let attr in video_attributes){
+    video.setAttribute(attr, video_attributes[attr]);
+  }
+  figure.appendChild(video);
+  figure.appendChild(figcaption);
+  return figure;
+}
+
+function displayStream(selector, stream) {
+  let video_element = document.querySelector(selector);
```

Comment:
@SingerDavid If check line 262, yr calling `displayStream(`#peer ID: ${id}`, stream);` with invalid ID. You'll adjust the string to make it work.